src/main/java/org/folio/dcb/service/impl/TransactionsServiceImpl.java

```diff
@@ -26,4 +34,28 @@ public TransactionStatusResponse createCirculationRequest(String dcbTransactionI
     };
   }
 
-}
+  public TransactionStatusResponse getTransactionStatusById(String dcbTransactionId) {
+    log.debug("getTransactionStatusById:: id {} ", dcbTransactionId);
+    TransactionEntity transactionEntity = getTransactionEntityOrThrow(dcbTransactionId);
+
+    return generateTransactionStatusResponseFromTransactionEntity(transactionEntity);
+  }
+
+  private TransactionStatusResponse generateTransactionStatusResponseFromTransactionEntity(TransactionEntity transactionEntity) {
+    TransactionStatus.StatusEnum transactionStatus = transactionEntity.getStatus();
+    DcbTransaction dcbTransaction = dcbTransactionMapper.mapToDcbTransaction(transactionEntity);
+    TransactionStatusResponse.StatusEnum transactionStatusResponseStatusEnum = TransactionStatusResponse.StatusEnum.fromValue(transactionStatus.getValue());
+
+    return TransactionStatusResponse.builder()
+      .status(transactionStatusResponseStatusEnum)
+      .item(dcbTransaction.getItem())
+      .patron(dcbTransaction.getPatron())
+      .build();
```

Comment:
As part of the response do we need item and patron details ? Will status enum be sufficient enough ?
Another thought maybe we might need the transaction role too in the response (coming near future).